python/eos/analysis_TEST.py

```diff
@@ -0,0 +1,80 @@
+import unittest
+import eos
+import numpy as np
+
+class ClassMethodTests(unittest.TestCase):
+
+    def test_parameter_scaling(self):
+
+        analysis_args = {
+            'global_options': { 'form-factors': 'BSZ2015', 'model': 'CKMScan' },
+            'priors': [
+                { 'parameter': 'CKM::abs(V_cb)',           'min':  38e-3, 'max':  45e-3 , 'type': 'uniform'},
+                { 'parameter': 'B->D::alpha^f+_0@BSZ2015', 'min':  0.0,   'max':  1.0   , 'type': 'uniform'},
+                { 'parameter': 'B->D::alpha^f+_1@BSZ2015', 'min': -4.0,   'max': -1.0   , 'type': 'uniform'},
+                { 'parameter': 'B->D::alpha^f+_2@BSZ2015', 'min': +4.0,   'max': +6.0   , 'type': 'uniform'},
+                { 'parameter': 'B->D::alpha^f0_1@BSZ2015', 'min': -1.0,   'max': +2.0   , 'type': 'uniform'},
+                { 'parameter': 'B->D::alpha^f0_2@BSZ2015', 'min': -2.0,   'max':  0.0   , 'type': 'uniform'}
+            ],
+            'likelihood': [
+                'B->D::f_++f_0@HPQCD2015A',
+                'B->D::f_++f_0@FNALMILC2015B',
+                'B^0->D^+e^-nu::BRs@Belle-2015A',
+                'B^0->D^+mu^-nu::BRs@Belle-2015A'
+            ]
+        }
+
+        # Test analysis definition
+        analysis = eos.Analysis(**analysis_args)
+
+        # Test analysis optimization
+        bfp = analysis.optimize()
+
+        # Test parameter scaling
+        point = bfp.point
+
+        self.assertTrue(
+            np.max(analysis._par_to_x(point)) < 1.0
+            )
+
+        self.assertTrue(
+            np.min(analysis._par_to_x(point)) > -1.0
+            )
+
+        self.assertTrue(
+            np.max(analysis._x_to_par(analysis._par_to_x(point)) - point) < 1e-10
+            )
+
+
+    def test_sanitize_manual_input(self):
+
+        types  = np.array(["Gaussian", "Gaussian"])
+        values = np.array([4.123, 3.141])
+
+        input_output_cases = [
+            {'dirty': np.float64(1.61), 'clean': float(1.61)},
+            {'dirty': np.str_('Gaussian'), 'clean': str("Gaussian")},
+            {
+             'dirty': {'type': types[0], 'mean': values[0]}, # Simplified manual contraint with numpy types
+             'clean': {'type': str('Gaussian'), 'mean': float(4.123)}
+            },
+            {
+             'dirty': {'type': types, 'mean': values}, # Simplified manual contraint with numpy arrays
+             'clean': {'type': [str('Gaussian'), str('Gaussian')], 'mean': [float(4.123), float(3.141)]}
+            },
+        ]
+
+        for case in input_output_cases:
+            # check value and type
+            self.assertEqual(
+                eos.Analysis._sanitize_manual_input(case['dirty']),
+                case['clean']
+            )
+            self.assertEqual(
+                type(eos.Analysis._sanitize_manual_input(case['dirty'])),
+                type(case['clean'])
+            )
```

Comment:
I believe the assertEquals are not performing a deep comparison of the dicts. At least for the type check, we only verify that a dict is going in and out.

We should in any case add the inverse check to also test our test method, i.e. check that the cleaned output is not equal to the dirty input, as long as we only check cases where we expect sanitisation.

Reply:
For the types I agree, that is why I am also testing `str` and `float`. For the second point I also agree, I will add this test

Reply:
Pffiou this was not trivial since you cannot unittest both the type and the value in one shot... I finally solved this by testing that the initial yaml strings were different while the sanitize yaml strings are the same. In the end this is even a better test since this is the aim of the helper function...